csp/templatetags/csp.py

```diff
@@ -19,10 +19,9 @@ def __init__(self, nodelist):
 
     def render(self, context):
         output = self.nodelist.render(context)
-        nonce = context.get('request')
+        request = context.get('request')
+        nonce = request.csp_nonce() if hasattr(request, 'csp_nonce') else ''
```

Comment:
I may be reading this backwards, but it looks like `process_request` is going to give you a hash no matter what... at the very least, it may give you an empty `request.csp_nonce`.
From here, it should just be a matter of grabbing the hash and stuffing it into a variable.
So this might work better as `nonce = getattr(request, 'csp_nonce', '')`

Sorry if I'm missing a step between the middleware engine and the template tag.